compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1007,6 +1008,14 @@ pub struct FreeRegionInfo {
     pub is_impl_item: bool,
 }
 
+#[derive(Copy, Clone)]
+pub struct TyCtxtFeed<'tcx> {
+    pub tcx: TyCtxt<'tcx>,
+    pub def_id: LocalDefId,
```

Comment:
this field should not be public. It should be read-only, otherwise one could create a def id and then change the def id of the `TyCtxtFeed`